base/dates/types.jl

```diff
@@ -169,21 +215,32 @@ calendar(dt::Date) = ISOCalendar
 """
     eps(::DateTime) -> Millisecond
     eps(::Date) -> Day
+    eps(::Time) -> Nanosecond
 
-Returns `Millisecond(1)` for `DateTime` values and `Day(1)` for `Date` values.
+Returns `Millisecond(1)` for `DateTime` values, `Day(1)` for `Date` values, and `Nanosecond(1)` for `Time` values.
 """
 Base.eps
 
 Base.eps(dt::DateTime) = Millisecond(1)
 Base.eps(dt::Date) = Day(1)
+Base.eps(t::Time) = Nanosecond(1)
 
 Base.typemax(::Union{DateTime,Type{DateTime}}) = DateTime(146138512,12,31,23,59,59)
 Base.typemin(::Union{DateTime,Type{DateTime}}) = DateTime(-146138511,1,1,0,0,0)
 Base.typemax(::Union{Date,Type{Date}}) = Date(252522163911149,12,31)
 Base.typemin(::Union{Date,Type{Date}}) = Date(-252522163911150,1,1)
+Base.typemax(::Union{Time,Type{Time}}) = Time(23,59,59,999,999,999)
+Base.typemin(::Union{Time,Type{Time}}) = Time(0)
+
 # Date-DateTime promotion, isless, ==
 Base.promote_rule(::Type{Date},x::Type{DateTime}) = DateTime
-Base.isless(x::Date,y::Date) = isless(value(x),value(y))
 Base.isless(x::DateTime,y::DateTime) = isless(value(x),value(y))
+Base.isless(x::Date,y::Date) = isless(value(x),value(y))
+Base.isless(x::Time,y::Time) = isless(value(x),value(y))
 Base.isless(x::TimeType,y::TimeType) = isless(promote(x,y)...)
 ==(x::TimeType,y::TimeType) = ===(promote(x,y)...)
+function ==(a::Time,b::Time)
+    return hour(a) == hour(b) && minute(a) == minute(b) &&
+        second(a) == second(b) && millisecond(a) == millisecond(b) &&
+        microsecond(a) == microsecond(b) && nanosecond(a) == nanosecond(b)
+end
```

Comment:
Ok, I was thinking we had a restricted range of allowed nanoseconds for the internal storage. The one problem I can see with this is dealing with Integer overflow:

```julia
julia> x = Time(Nanosecond(typemax(Int64)))
23:47:16.854775807

julia> x + Nanosecond(1)
00:12:43.145224192
```